### PR TITLE
Issue #5, Create Item Class

### LINKED ISSUE #5: Create Item class
### Description
As food items are the core object that this project will manage, it will be a good starting point.
The Item class should include member variables for

- Name of item
- Price
- Availability/Stock of item
- Potential sales or specials that an item can have, and whether they are in effect. 

This class should be as simple as possible, and it should be easy to interact with in other classes.

### PR DESCRIPTION
**Pull request by Jahrr**
This PR adds the Item class, which contains variables for ID, item name, price, stock of item, and additional information.
closes #5

### CODE REVIEW

File: src/Item.hpp
Comment (Kelarov) [inline src/Item.hpp:1]: Hi. Is there a reason you included std::vector, but didn't really use it?

File: bin/CMakeLists.txt
Comment (Kelarov) [inline bin/CMakeLists.txt:1]: 1. Couldn't we set a SOURCE_FILES variable and pass it to the `add_executable()` command instead?
Though I think the best option would be to use `target_include_directories` and use `#include<>` to refer to the header files inside the .cpp's. 
2. `./src` might become quite crowded with header and source files; we could create an `inc` directory for the headers and use it for `target_include_directories`

**Review by Kelarov — COMMENTED** [review]
Looks good!

File: src/Item.hpp
Comment (Jahrr) [reply src/Item.hpp:1]: Ah good catch I think I was planning to use it for allergens/ingredients but forgot to erase it after I scrapped the idea in favor of the additional information variable. It should be taken out.

File: bin/CMakeLists.txt
Comment (Jahrr) [reply bin/CMakeLists.txt:1]: Good idea but its out of the scope of this PR. Refactoring and changing how we build should be an issue of its own.

Comment (Kelarov) [reply bin/CMakeLists.txt:1]: Yeah, I think this is a fair point.

File: src/Item.cpp
Comment (genciberisha) [inline src/Item.cpp:12]: item_ID_{item_ID};

File: bin/bluey.cpp
Comment (genciberisha) [inline bin/bluey.cpp:2]: Move your .hpp files to **<Project_root>/include/sudoManager**.
Use `target_include_directories()`.

This line should look like this: `#include "sudoManager/Item.hpp"`

File: bin/CMakeLists.txt
Comment (genciberisha) [reply bin/CMakeLists.txt:1]: This PR contains the base code from which we will build on.
Leaving the `/CMakeLists.txt` like this might create problems in the future.
Please fix this.

**Comment by genciberisha** [comment]
Squash all commits into one and add a meaningful name.

**Comment by genciberisha** [comment]
Since we do not have a workflow that automatically formats the code with the `.clang-format` config.
I think you should format the code before submitting it.

We are gonna add that workflow soon.

File: src/Item.cpp
Comment (RockyDesigne) [reply src/Item.cpp:12]: I think there should only be one constructor, and all the parameters should have optional values, because that allows for greater flexibility, and std::string_view should be used. Also the member variables in the initializer list are 
initialized in the order in which they are declared in the class, so they should match that order.
Here the explicit keyword is used to avoid implicit conversions.
`
explicit Item(std::string_view name = "", int item_ID = 0, float price = 0.0, int stock = 0)
        : name_ {name}, item_ID_ {item_ID}, price_ {price}, stock_ {stock} {}
        `

File: src/Item.cpp
Comment (RockyDesigne) [inline src/Item.cpp:41]: For printing the item info we should probably overload operator<< as a friend function something like this:
`
friend std::ostream& operator<<(std::ostream& os, const Item& item) {
         os << item.name_ << " " << item.item_ID_ << " " << item.price_ << " " << item.stock_;
         return os;
     }
`

**Comment by Jahrr** [comment]
Please merge or suggest changes so we can continue development

File: src/Item.cpp
Comment (Kelarov) [reply src/Item.cpp:45]: @Jahrr do we have an overload for the `<<` operator as @RockyDesigne suggested, or are we gonna leave it like it is?

**Review by Kelarov — COMMENTED** [review]
- Waiting for response-comment about the `printItemInfo()`
- @genciberisha responsible for approving the PR

**Review by Kelarov — COMMENTED** [review]
Resolve `CMakeLists.txt` conflict

File: src/Item.cpp
Comment (Jahrr) [reply src/Item.cpp:45]: It seems unnecessary to overload the operator in this case. If we ever want to print to a file we can use std::stringstream. I'm going to leave it as is to keep it simple and understandable for an inexperienced reader, and if we need to we can change this in the future.

File: src/Item.cpp
Comment (Jahrr) [reply src/Item.cpp]: String_view is now used, but the two constructors were kept to allow ease of object creation.

**Comment by genciberisha** [comment]
@Jahrr [Making a good pull request item Nr.5](https://github.com/NextPP/sudoManager/blob/main/CONTRIBUTING.md)
Before merging there should be only one commit with a meaningful name.

**Review by genciberisha — CHANGES_REQUESTED** [review]

**Pull request closed without merging**
